skiko/src/jvmMain/kotlin/org/jetbrains/skiko/FrameWatcher.kt

```diff
@@ -0,0 +1,35 @@
+package org.jetbrains.skiko
+
+import kotlinx.coroutines.*
+import java.util.concurrent.atomic.AtomicInteger
+
+/**
+ * This class is intended to mitigate issues coming from the situation that we have
+ * pretty large native peers (Skia objects) for rather tiny Java wrappers.
+ * It is especially visible for situation with paragraph classes.
+ * As a result, memory consumption grows dramatically.
+ * To solve this issue, we force periodic GC if certain amount of frames was rendered,
+ * making sure that we'll have memory consumption under control.
+ */
+internal object FrameWatcher {
+    fun start() {
+        // We initiate GC on IO threads, so that rendering is not blocked in
+        // cases of concurrent collectors.
+        @OptIn(DelicateCoroutinesApi::class)
+        GlobalScope.launch(Dispatchers.IO) {
+            while (true) {
+                delay(5000L)
```

Comment:
Maybe instead of delay we will use something like frameCountChannel?

So there will be no background daemon eating CPU (not sure how it affects battery on laptops)

Reply:
We could use `wait() / notify()` instead of `delay` and inside the `nextFrame()` function check the counter and notify if `frameCounter > 1000.

Reply:
Guess with this kind of update frequency it's not too important which mechanism will be used. 30 seconds is not very frequent. Will merge current solution to make a build for customers, and then can work on improvements, if will be needed.